backend/src/api/billing/billing.api.ts

```diff
@@ -0,0 +1,89 @@
+import { FastifyPluginAsync, FastifyRequest } from 'fastify';
+
+import { BillingApiPath, HttpCode, HttpMethod } from '~/common/enums/enums';
+import {
+  BillingReplenishParamsDto,
+  UserGetResponseWithMoneyBalanceDto,
+} from '~/common/types/types';
+import {
+  billing as billingService,
+  user as userService,
+  userDetails as userDetailsService,
+} from '~/services/services';
+import { billingReplenishParams as billingReplenishParamsValidationSchema } from '~/validation-schemas/validation-schemas';
+
+type Options = {
+  services: {
+    billing: typeof billingService;
+    user: typeof userService;
+    userDetails: typeof userDetailsService;
+  };
+};
+
+const initBillingApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
+  const {
+    billing: billingService,
+    user: userService,
+    userDetails: userDetailsService,
+  } = opts.services;
+
+  fastify.route({
+    method: HttpMethod.POST,
+    url: BillingApiPath.REPLENISH,
+    schema: { body: billingReplenishParamsValidationSchema },
+    async handler(
+      req: FastifyRequest<{ Body: BillingReplenishParamsDto }>,
+      rep,
+    ) {
+      const { id } = req.user;
+      const { amountOfMoneyToReplenish } = req.body;
+      const userWithBalance = await userService.getByIdWithMoneyBalance(id);
+
+      const replenishDto = await billingService.initReplenish(
+        amountOfMoneyToReplenish,
+      );
+
+      let userDetailsWithBalance = null;
+
+      if (replenishDto.status === 'complete') {
+        const newBalance =
+          (userWithBalance as UserGetResponseWithMoneyBalanceDto).userDetails
+            .moneyBalance + amountOfMoneyToReplenish;
+        userDetailsWithBalance = await userDetailsService.updateMoneyBalance(
+          id,
+          newBalance,
+        );
+      }
+
+      rep.status(HttpCode.OK).send(userDetailsWithBalance);
+    },
+  });
+
+  fastify.route({
+    method: HttpMethod.POST,
+    url: BillingApiPath.WITHDRAW,
```

Comment:
Let's add a validation schema for this route

Reply:
There is nothing to validate for now)

Reply:
Oh, my bad :)
Are you withdrawing all the money? Is there a possibility to withdraw just some requested amount?

Reply:
Yes, we are withdrawing all money. For now, I think it is good approach